cmd/argocd/commands/app.go

```diff
@@ -675,7 +675,7 @@ func NewApplicationUnsetCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 				}
 				for _, env := range pluginEnvs {
 					err = app.Spec.Source.Plugin.RemoveEnvEntry(env)
-					if err != nil {
```

Comment:
@alexmt  Found this while testing against the latest RC. I think #6457 introduced this change. The env vars will not be removed until there's an error